<launch>

  <!-- This is the main state machine for interaction and PbD -->
  <arg name="isReload" value="False" />
  <arg name="dataRoot" value="$(env HOME)" />
  <arg name="experimentNumber" value="3" />
  
  <!-- This runs the interactive manipulation -->
  
  <node pkg="pr2_arm_kinematics" type="pr2_arm_kinematics_node" name="pr2_left_arm_kinematics_simple" output="screen">
   <param name="tip_name" value="l_wrist_roll_link" />
   <param name="root_name" value="torso_lift_link" />
   <param name="maxIterations" value="10000" />
  </node>

  <node pkg="pr2_arm_kinematics" type="pr2_arm_kinematics_node" name="pr2_right_arm_kinematics_simple" output="screen">
   <param name="tip_name" value="r_wrist_roll_link" />
   <param name="root_name" value="torso_lift_link" />
   <param name="maxIterations" value="10000" />
  </node>
  
  <include file="$(find pr2_interactive_manipulation)/launch/pr2_interactive_manipulation_robot.launch" >
    <arg name="sim" value="false" />
  </include>

  
  <node name="pr2_pbd_interaction" pkg="pr2_pbd_interaction" type="interaction.py" output="screen">
    <param name="dataRoot" value="$(arg dataRoot)" />
    <param name="isReload" value="$(arg isReload)" />
    <param name="experimentNumber" value="$(arg experimentNumber)" />
  </node>
  
  <!-- This makes the robot look around appropriately -->

  <include file="$(find pr2_social_gaze)/launch/gaze.launch"/>

  <!-- This is needed for filling in the velocity of replayed trajectories -->

  <node pkg="trajectory_filter_server" name="trajectory_filter" type="trajectory_filter_server" output="screen">
   <rosparam command="load" file="$(find trajectory_filter_server)/config/joint_limits.yaml"/>
   <rosparam command="load" file="$(find pr2_pbd_interaction)/config/filters.yaml"/>
  </node>

  <include file="$(find r_track_alvar)/launch/pr2_indiv.launch"/>
</launch>
